mono/metadata/exception.c

```diff
@@ -596,7 +600,8 @@ mono_get_exception_type_initialization (const gchar *type_name, MonoException *i
 	args [0] = mono_string_new (mono_domain_get (), type_name);
 	args [1] = inner;
 
-	exc = mono_object_new (mono_domain_get (), klass);
+	exc = mono_object_new_checked (mono_domain_get (), klass, &error);
```

Comment:
This factory methods for exception objects are really messed up, they use a runtime_invoke that can throw <o>.